### DISCUSSION TITLE
Can't resolve 'async_hooks'

### DISCUSSION THREAD

**Discussion by melloware** [nodejs/undici#1057] [Q&A] [answered]
Your library is used through another library I use https://github.com/lbdremy/solr-node-client Solr Client.  I cannot get my app to build with this error

```
./node_modules/undici/lib/api/api-request.js
Module not found: Can't resolve 'async_hooks' in 'C:\dev\aim\node_modules\undici\lib\api'
```

And looking at your package.json I don't see anywhere you include `async_hooks`???

when I try `npm run build` i am getting this different error

```
Creating an optimized production build...
Failed to compile.

.\node_modules\undici\lib\fetch\request.js
Cannot find module: 'stream/web'. Make sure this package is installed.
```

**Comment by mcollina** [comment, marked as answer]
Those are node core modules and should not be bundled. I think you have some misconfiguration in your bundling/transpiling setup.

(I recommend to not bundle the server side)

**Reply by melloware** [reply]
Ok interesting I am just using solar-insight in my package.json.  If I back up to their version 0.9.0 version it works fine. So is it something wrong with SOLR Insights build?

**Reply by mcollina** [reply]
You should open an issue there.

**Reply by melloware** [reply]
@mcollina if you look below @kibertoad is one of the solr-client developers and he has a question.

**Comment by kibertoad** [comment]
@mcollina solr-node-client started depending on undici. Does undici need async_hooks to work? Wonder if it is related to diagnostic channels and could be handled with fallbacks when not available.

**Reply by mcollina** [reply]
Undici needs `async_hooks` to run. It supports `async_hooks` to keep the `AsyncLocalStorage` work across multiple requests. `async_hooks` are available in all Node.js version Undici supports, so I'm a bit unsure why it's not there.

**Reply by kibertoad** [reply]
Because webpack/frontend. Gotcha, we'll implement fallback for such cases.

**Reply by mcollina** [reply]
I think you should not be bundling undici at all, it's not a frontend library.

**Reply by kibertoad** [reply]
Well yeah, hence my idea is to fallback to using axios in case solr-node-client detects it is being used in browser environment.
